driver/src/price_finding/optimization_price_finder.rs

```diff
@@ -237,54 +238,65 @@ fn deserialize_result(result: String) -> Result<models::Solution> {
 }
 
 impl PriceFinding for OptimisationPriceFinder {
-    fn find_prices(
-        &self,
-        orders: &[models::Order],
-        state: &models::AccountState,
+    fn find_prices<'a>(
+        &'a self,
+        orders: &'a [models::Order],
+        state: &'a models::AccountState,
         time_limit: Duration,
-    ) -> Result<models::Solution> {
-        let input = solver_input::Input {
-            tokens: self.price_oracle.get_token_prices(&orders).wait(),
-            ref_token: TokenId(0),
-            accounts: serialize_balances(&state, &orders),
-            orders: orders.iter().map(From::from).collect(),
-            fee: self.fee.as_ref().map(From::from),
-        };
-
-        let now = Utc::now();
-        // We are solving the batch before the current one
-        let batch_id = (now.timestamp() / 300) - 1;
-        let date = now.format("%Y-%m-%d");
-
-        let input_folder = format!("instances/{}", &date);
-        let input_file = format!(
-            "{}/instance_{}_{}.json",
-            &input_folder,
-            &batch_id,
-            &now.to_rfc3339()
-        );
+    ) -> BoxFuture<'a, Result<models::Solution>> {
+        let price_oracle = &*self.price_oracle;
```

Comment:
If we do `self.price_oracle.get_token_prices` as it used to be but in the async block then the compiler complains that `self.price_oracle` isn't `Send`.